### ISSUE TITLE
tcp_listen_options is causing clients not to be able to connect

### ISSUE THREAD

**Issue by internaught** [voxpupuli/puppet-rabbitmq#719]
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 4.5.2
- Ruby:
- Distribution: CentOS 7
- Module version: 8.2.0

## How to reproduce (e.g Puppet code you use)
use this module 

## What are you seeing
Clients aren't connecting

## What behaviour did you expect instead
i expect clients to be able to connect

## Output log
[E 180830 21:31:31 events:123] Failed to capture events: '[Errno 104] Connection reset by peer', trying again in 4 seconds.

## Any additional information you'd like to impart
There should be an option to not include this part of the config:

```
    {tcp_listen_options, [
         {backlog,       128},
         {nodelay,       true},
         {linger,        {true, 0}},
         {exit_on_close, false}
    ]},
```

Having these options in the config makes clients fail to connect. Removing these lines from the config, clients can connect fine. Please add an option to be able to remove these lines from the config as these seem to not be sane defaults.

setting `config_ranch => false` seems to have solved my issue, please update the readme to explain this option.

**Comment by wyardley** [comment]
#621 has some more background and discussion about this
If memory serves, we intended to avoid making the behavior too "magical" (and keep it idempotent) by basing it on the rabbitmq version fact.

If I get a chance, I'll add a note somewhere in the README and also document that param in the docstrings.

**Comment by wyardley** [comment]
I've created a docs fix for this in #719 -- hope this helps.
